src/consts/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React, { ReactNode, useLayoutEffect, useState } from 'react'
+import { createBrowserHistory } from 'history'
+import { Router } from 'react-router-dom'
+
+export const history = createBrowserHistory({ window })
+
+interface HubRouterProps {
+    children?: ReactNode
+}
+
+export function HubRouter(props: HubRouterProps) {
```

Comment:
We need a custom `Router` because… there's no safe way of accessing the `history` object within `BrowserRouter`, and `history` objects created by `createBrowserHistory` utility are in no way linked. A singleton would be nice, eh?